### PR TITLE
Build fabtests and run runfabtests.sh

### PR DESCRIPTION
**Pull request by shantonu**
After successfully building libfabric:
- clone fabtests
- build fabtests
- run fabtests over sockets provider

Signed-off-by: Shantonu Hossain <shantonu.hossain@intel.com>

### CODE REVIEW

**Comment by shefty** [comment]
Please see if there's a way to do this using container support, without invoking sudo

**Comment by shefty** [comment]
Please see fabtests PR #368

**Comment by raffenet** [comment]
The librdmacm packages should be easy enough to install to a non-root location by changing prefix. For the other `apt-get install` commands, you could convert them to `apt-get source` then install again to a common non-root location.

**Comment by shefty** [comment]
We should decide if we want to pull down the packages or use the git repos or tar files

**Comment by shantonu** [comment]
The Ubuntu version they have has older version of libibverbs and librdmacm. I think that's why Ken added the step to pull down latest packages.

**Comment by shefty** [comment]
Since we're only looking at building the other providers, I'm leaning toward using the latest git tree.  This does leave the potential that our checks will fail if the upstream repos are broken.  The alternative is to pull a specific tag or tarfile, which means we would need to manually update the scripts to use newer versions.

Is there some best practice that anyone knows of for this?

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,18 @@
-sudo: false
```
Comment (shefty) [inline .travis.yml]: We want to keep this option.  It allows the tests to run quicker.

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+env:
+        - CFLAGS=-I${TRAVIS_BUILD_DIR}/install/include LDFLAGS=-L${TRAVIS_BUILD_DIR}/install/lib LD_LIBRARY_PATH=${TRAVIS_BUILD_DIR}/install/lib
 install: ./autogen.sh
+before_script:
+        - sudo apt-get update
+        - sudo apt-get install libnl-dev
+        # the system provided libibverbs and rdmacm is too old
+        - pushd /tmp && wget https://www.openfabrics.org/downloads/verbs/libibverbs-1.1.8.tar.gz && tar xvfz libibverbs-1.1.8.tar.gz && cd libibverbs-1.1.8 && ./configure --prefix=${TRAVIS_BUILD_DIR}/install && make -j && make install && popd
+        - pushd /tmp && wget https://www.openfabrics.org/downloads/rdmacm/librdmacm-1.0.21.tar.gz && tar xvfz librdmacm-1.0.21.tar.gz && cd librdmacm-1.0.21 && ./configure --prefix=${TRAVIS_BUILD_DIR}/install CC=gcc && make -j && make install && popd
```
Comment (shefty) [inline .travis.yml]: Use multiple lines for readability, versus one really, really, really, really, really long line

Comment (shefty) [inline .travis.yml]: It seems undesirable to hard-code a specific version of these packages.

File: .travis.yml (hunk 1 above)
Comment (shantonu) [inline .travis.yml]: Yes, I removed the sudo command as @raffenet mentioned and added back this line.

File: .travis.yml (hunk 2 above)
Comment (shantonu) [inline .travis.yml]: Ok, I'll use the respective repos directly!

**Comment by shantonu** [comment]
Closing this PR to cleanup

**Pull request closed without merging**